src/test/regress/sql/with.sql

```diff
@@ -1083,3 +1083,29 @@ WITH cte AS (
 
 RESET optimizer;
 DROP TABLE d;
+
+-- Check if sharing is disabled for a SegmentGeneral CTE to avoid deadlock if CTE is
+-- executed with 1-gang and joined with n-gang
+SET optimizer = off;
+--start_ignore
+DROP TABLE IF EXISTS d;
```

Comment:
maybe it's worth reusing the table from the previous test? and maybe also don't do reset optimizer so that you don't turn it off again later?

Reply:
I think the tests should be self-sufficient, not rely on previous ones and avoid any side effects. What if someone writes a new test between the mentioned two and messes the data in it?

`RESET` sets to default value.